common/lua-system.cpp

```diff
@@ -6,25 +6,64 @@
 #include <selene.h>
 
 #include "entity.hpp"
+#include "events.hpp"
+#include "multiton.hpp"
+
+namespace tec {
+	using LuaScriptMap = Multiton<eid, LuaScript*>;
 
-namespace tec {	
 	void LuaSystem::Update(const double delta) {
 		auto _log = spdlog::get("console_log");
 		ProcessCommandQueue();
-		
-		for (auto itr = this->scripts_map.Begin(); itr != this->scripts_map.End(); itr++) {
+		EventQueue<EntityCreated>::ProcessEventQueue();
+		EventQueue<EntityDestroyed>::ProcessEventQueue();
+
+		for (auto itr = LuaScriptMap::Begin(); itr != LuaScriptMap::End(); itr++) {
 			auto entity_id = itr->first;
```

Comment:
unused variable

Reply:
A result of C&P from other systems, will remove.

Reply:
For now I have commented this out. This area of the code will be revisited when we rework the lua stuff.